src/hotspot/share/jfr/recorder/repository/jfrEmergencyDump.cpp

```diff
@@ -372,21 +372,26 @@ static void write_repository_files(const RepositoryIterator& iterator, char* con
     assert(fqn != NULL, "invariant");
     current_fd = open_exclusivly(fqn);
     if (current_fd != invalid_fd) {
-      const int64_t size = file_size(current_fd);
+      const size_t size = (size_t)file_size(current_fd);
```

Comment:
There's an existing bug here: error code of -1 from `file_size` is not handled.

Reply:
The type is returned to signed and the assert checks the -1.